### PR TITLE
Diagnostics

### PR DESCRIPTION
**Pull request by raoulcollenteur**
# Short Description
This PR updates the stats module in numerous ways and is accompanied by an example notebook about diagnostic checking of Pastas models.

**New:**
- methods in ps.stats: `diagnostics, plot_diagnostics, stoffer_toloi, plot_acf`
- metrics.py: all goodness-of-fit metrics may now be used separately (e.g., ps.stats.evp)
- 4_diagnostic_checking.ipynb 

**Updated:**
- ml.plots.diagnostics
- methods in tests.py: `ljung_box, durbin_watson, runs_test`

Documention for this branch is also compiled and visible here:
https://pastas.readthedocs.io/en/diagnostics/api/statistics.html

# Checklist before PR can be merged:
- [X] is documented
- [X] PEP8 compliant code
- [X] tests added / passed
- [X] passes Travis
- [X] Example Notebook (for new features)

### CODE REVIEW

**Comment by raoulcollenteur** [comment]
The Notebook requires some more work but I think we can do this in the Dev-branch so we also include all changes there when working on the notebook.

File: pastas/stats/tests.py [hunk 1]
```diff
@@ -227,52 +244,292 @@ def runs_test(series, alpha=0.05, cutoff="mean"):
...
+    Returns
+    -------
+    qm: float
+        Adapted Ljung-Box test statistic.
+    pval: float
+        p-value for the test statistic, based on a chi-squared distribution.
 
+    Notes
+    -----
+    Portmanteau test can handle missing data (nan's) in the input time series.
```
Comment (dbrakenhoff) [inline pastas/stats/tests.py:281]: Change name to stoffer-toloi?

File: pastas/stats/tests.py [hunk 2]
```diff
@@ -227,52 +244,292 @@ def runs_test(series, alpha=0.05, cutoff="mean"):
...
+                      Checks Statistic P-value  Reject H0
+    Shapiroo       Normality      1.00    0.86      False
+    D'Agostino     Normality      1.18    0.56      False
+    Runs test      Autocorr.     -0.76    0.45      False
+    Durbin-Watson  Autocorr.      2.02     nan      False
+    Ljung-Box      Autocorr.      5.67    1.00      False
+
+    Note
+    ----
+    This plot assumed that the noise or residuals follow a Normal
```
Comment (dbrakenhoff) [inline pastas/stats/tests.py:487]: what is this note referring to? Which plot? assumed=assumes?

**Comment by raoulcollenteur** [comment]
I thought this makes more sense as it is really part of the stats subpackage right now. One could do:

from pastas import stats

and do a diagnostic testing of non-Pastas models as well :)

Merging.

**Pull request merged**